src/controller/CommissionerDiscoveryController.cpp

```diff
@@ -135,55 +210,178 @@ void CommissionerDiscoveryController::Ok()
+        return;
+    }
+    if (client->GetUDCClientProcessingState() != UDCClientProcessingState::kObtainingOnboardingPayload)
+    {
+        ChipLogError(AppServer, "UX Ok - HandleContentAppCheck: invalid state for HandleContentAppPasscodeResponse");
+        return;
+    }
 
-        CharSpan rotatingIdSpan = chip::CharSpan(rotatingIdString, sizeof(rotatingIdString));
-        uint32_t passcode       = 0;
-        uint8_t targetAppCount  = client->GetNumTargetAppInfos();
-        if (targetAppCount > 0)
+    uint8_t targetAppCount = client->GetNumTargetAppInfos();
+    for (uint8_t i = 0; i < targetAppCount; i++)
+    {
+        TargetAppInfo info;
+        if (client->GetTargetAppInfo(i, info))
         {
-            bool hasTargetApp = false;
-            for (uint8_t i = 0; i < targetAppCount; i++)
+            if (info.checkState == TargetAppCheckState::kAppFoundPasscodeReturned)
             {
-                TargetAppInfo info;
-                if (client->GetTargetAppInfo(i, info))
+                // nothing else to do, complete commissioning has been called
+                return;
+            }
+            else if (info.checkState == TargetAppCheckState::kAppFoundNoPasscode)
+            {
+                foundTargetApp = true;
+            }
+            else if (info.checkState == TargetAppCheckState::kNotInitialized)
+            {
+                if (target.vendorId == info.vendorId && target.productId == info.productId)
                 {
-                    if (mPasscodeService->HasTargetContentApp(client->GetVendorId(), client->GetProductId(), rotatingIdSpan, info,
-                                                              passcode))
+                    client->SetTargetAppInfoState(i, target.checkState);
+                    if (target.checkState != TargetAppCheckState::kAppNotFound)
                     {
-                        // found one
-                        hasTargetApp = true;
+                        foundTargetApp = true;
                     }
                 }
-            }
-            // handle NoAppsFound CDC case
-            if (!hasTargetApp && client->GetNoPasscode())
-            {
-                ChipLogError(AppServer, "UX Ok: target apps specified but none found, sending CDC");
-                CommissionerDeclaration cd;
-                cd.SetNoAppsFound(true);
-                mUdcServer->SendCDCMessage(
-                    cd, chip::Transport::PeerAddress::UDP(client->GetPeerAddress().GetIPAddress(), client->GetCdPort()));
-                return;
+                else
+                {
+                    foundPendingTargets = true;
+                }
             }
         }
-        else
-        {
-            passcode = mPasscodeService->FetchCommissionPasscodeFromContentApp(client->GetVendorId(), client->GetProductId(),
-                                                                               rotatingIdSpan);
-        }
+    }
+    if (passcode != 0)
```

Comment:
I did not review the logic changes here carefully.  I'm not sure why there are logic changes here at all; they seem unrelated to teh "remove sync calls" bit.

Reply:
the logic did not change but in some cases functions had to refactored so that multiple lookups could be done (each with its own callback)